swim/memberlist.go

```diff
@@ -166,10 +166,10 @@ func (m *memberlist) RandomPingableMembers(n int, excluding map[string]bool) []*
 }
 
 // returns an immutable slice of members representing the current state of the membership
-func (m *memberlist) GetMembers() (members []Member) {
+func (m *memberlist) GetMembers() (members []*Member) {
 	m.members.RLock()
 	for _, member := range m.members.list {
-		members = append(members, *member)
+		members = append(members, member)
```

Comment:
Good point. Is it unsafe though? We need to answer this question definitively.

There are two things here that have mutexes:

* `m.members`: an anonymous struct on the memberlist. Presumably this lock is for mutating the list.
* `Member`s themselves. Presumably, this lock is for changing the status or incarnation number of the member.

`GetMembers()` returns a new slice containing pointers to the members. I can only see this used in two places. In both cases, data from the member is being read without acquiring a read lock.

One is for generating stats and the other place is for creating a list of changes for a full sync. Might we be OK to just change these places to acquire the read lock on the member before reading data?